src/lib/i18n/index.es

```diff
@@ -0,0 +1,211 @@
+/**
+ * Default language.
+ * @type {string}
+ */
+const DEFAULT_LANGUAGE = 'en'
+
+/**
+ * Pattern to detect params in a string.
+ * @type {Regexp}
+ */
+let paramsPattern = /%(\d+)/g
+
+/**
+ * To store all the created instances.
+ * @type {Object}
+ */
+let instances = {}
+
+/**
+ * Current language. Shared by all the instances.
+ * @type {String}
+ */
+let language = DEFAULT_LANGUAGE
+
+/**
+ * Public API.
+ * @type {Object}
+ */
+class I18N {
+
+  /**
+   * Creates an i18n instance.
+   * If the ID is specified returns a singleton for each ID.
+   * If the ID is not specified returns a new instance.
+   * @param {string} id Instance ID. Optional.
+   */
+  constructor (id) {
+    let isValidId = typeof id === 'string'
+
+    if (!!isValidId && !!instances[id]) {
+      return instances[id]
+    } else if (!!isValidId && !instances[id]) {
+      instances[id] = this
+    }
+
+    /**
+     * Object where the translations are stored.
+     * @type {Object}
+     */
+    this._translations = {}
+  }
+
+  /**
+   * Sets the translations for a language.
+   * @param {string} lang Language identifier.
+   * @param {object} translations
+   * @returns {boolean} true if all went ok. false if not.
+   */
+  translations (lang, translations) {
+    if (!this._isValidLanguage(lang) || !this._areValidTexts(translations)) {
+      console.warn('i18n.translations: Invalid language or translations.')
```

Comment:
Don't save lines by grouping errors messages. If you use "err1 or err2", "err1 or err2 or err3 or err4" would be ok too?
Also, when you get the error message you would know exactly what is failing, otherwise you would waste your time debugging.